### PR TITLE
Ensure TermsAggregation size is never 0

### PR DESCRIPTION
**Pull request by AmitPhulera**
## Product Description

Starting from ES 5, the terms aggregation cannot have a size 0. This is not being used in HQ, except at 1 place where it was added by mistake. The PR ensures that we never set 0 in aggregation size.

For more context on why size 0 was removed - https://github.com/elastic/elasticsearch/issues/18838

## Safety Assurance
Pretty safe, no actual usage on HQ, just an additional check with tests
### Automated test coverage

Tests are there

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/users/views/utils.py [hunk 1]
```diff
@@ -144,7 +144,8 @@ def _get_location_case_counts(domain, location_ids):
         CaseES()
         .domain(domain)
         .owner(location_ids)
-        .aggregation(TermsAggregation('by_location', 'owner_id').size(0))
+        .aggregation(TermsAggregation('by_location', 'owner_id'))
+        .size(0)
```
Comment (esoergel) [inline corehq/apps/users/views/utils.py:148]:
```suggestion
```
I believe this `size` call is redundant, as we set size to 0 whenever aggregations are used anyways:

https://github.com/dimagi/commcare-hq/blob/2acb74f785ca8971e7ed99a39c745931272e12ca/corehq/apps/es/es_query.py#L384-L385

**Review by esoergel — APPROVED** [review]
good find!

File: corehq/apps/users/views/utils.py (hunk 1 above)
Comment (AmitPhulera) [reply corehq/apps/users/views/utils.py]: Thanks for the suggestion. This is nice 👍

**Review by esoergel — APPROVED** [review]

File: corehq/apps/es/aggregations.py [hunk 2]
```diff
@@ -215,6 +215,7 @@ class TermsAggregation(Aggregation):
     def __init__(self, name, field, size=None, missing=None):
         assert re.match(r'\w+$', name), \
             "Names must be valid python variable names, was {}".format(name)
+        assert size != 0, "Aggregation size must be greater than 0"
```
Comment (millerdev) [inline corehq/apps/es/aggregations.py:218]: Nit: Is "must be greater than 0" the actual requirement?
```suggestion
        assert size > 0, "Aggregation size must be greater than 0"
```
Or maybe the language should be changed?
```suggestion
        assert size != 0, "Aggregation size must not be zero"
```

File: corehq/apps/es/aggregations.py [hunk 3]
```diff
@@ -237,6 +238,7 @@ def order(self, field, order="asc", reset=True):
         return query
 
     def size(self, size):
+        assert size != 0, "Aggregation size must be greater than 0"
```
Comment (millerdev) [inline corehq/apps/es/aggregations.py:241]: Same as above.

File: corehq/apps/es/aggregations.py (hunk 2 above)
Comment (AmitPhulera) [reply corehq/apps/es/aggregations.py:218]: size can be `None` which means we can't compare it with an integer. So we can't use first suggestion.

Maybe I am overthinking it "must not be zero" would include negative numbers which are not supported. I am okay updating the language if you feel strongly about it.

Comment (millerdev) [reply corehq/apps/es/aggregations.py:218]: Maybe this then?
```suggestion
        assert size is None or size > 0, "Aggregation size must be greater than 0"
```

File: corehq/apps/es/aggregations.py (hunk 3 above)
Comment (AmitPhulera) [reply corehq/apps/es/aggregations.py]: [ae30a37](https://github.com/dimagi/commcare-hq/pull/33146/commits/ae30a37373e18393cb51058995433baf21729e6a)

File: corehq/apps/es/aggregations.py (hunk 2 above)
Comment (AmitPhulera) [reply corehq/apps/es/aggregations.py]: [ae30a37](https://github.com/dimagi/commcare-hq/pull/33146/commits/ae30a37373e18393cb51058995433baf21729e6a)

**Review by millerdev — APPROVED** [review]

**Pull request merged**